### PR TITLE
Play 2: Adjust thread pool/db pool size to CPU's available

### PR DESCRIPTION
**Pull request by mkurz**
By passing in the cpu count/thread count as Java system properties to the Play 2 apps we can configure the thread pools and db connection pools accordingly.
See https://github.com/TechEmpower/FrameworkBenchmarks/issues/4336#issuecomment-460355569

I also do some calculations for the `db_pool_size` property, so it is `((physical_core_count * 2) + effective_spindle_count)`, see https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing

All the values will be logged at application startup, so we can check if fetching the cpu/thread count worked out ok.

Let's see what Travis says.

### CODE REVIEW

**Comment by mkurz** [comment]
Ready to merge.

File: frameworks/Java/play2-java/play2-java-ebean-hikaricp.dockerfile [hunk 1]
```diff
@@ -5,4 +5,4 @@ COPY play2-java-ebean-hikaricp .
 RUN sed -i 's/.enablePlugins(PlayJava, PlayEbean, PlayNettyServer)/.enablePlugins(PlayJava, PlayEbean).disablePlugins(PlayNettyServer)/g' build.sbt
 
 RUN sbt stage
-CMD ["target/universal/stage/bin/play2-java-ebean-hikaricp", "-Dplay.server.provider=play.core.server.AkkaHttpServerProvider", "-J-server", "-J-Xms1g", "-J-Xmx1g", "-J-XX:NewSize=512m", "-J-XX:+UseG1GC", "-J-XX:MaxGCPauseMillis=30", "-J-XX:-UseBiasedLocking", "-J-XX:+AlwaysPreTouch"]
+CMD target/universal/stage/bin/play2-java-ebean-hikaricp -Dplay.server.provider=play.core.server.AkkaHttpServerProvider -J-server -J-Xms1g -J-Xmx1g -J-XX:NewSize=512m -J-XX:+UseG1GC -J-XX:MaxGCPauseMillis=30 -J-XX:-UseBiasedLocking -J-XX:+AlwaysPreTouch -Dthread_count=$(nproc) -Dphysical_cpu_count=$(grep ^cpu\\scores /proc/cpuinfo | uniq | awk '{print $4}') -Ddb_pool_size=$(( $(( $(grep ^cpu\\scores /proc/cpuinfo | uniq | awk '{print $4}') * 2 )) + 1 ))
```
Comment (msmith-techempower) [inline frameworks/Java/play2-java/play2-java-ebean-hikaricp.dockerfile]: Yikes

Comment (mkurz) [reply frameworks/Java/play2-java/play2-java-ebean-hikaricp.dockerfile]: You don't like it?

Comment (msmith-techempower) [reply frameworks/Java/play2-java/play2-java-ebean-hikaricp.dockerfile]: Haha, if I didn't like it, I wouldn't have merged it... that said... yikes.

Comment (mkurz) [reply frameworks/Java/play2-java/play2-java-ebean-hikaricp.dockerfile]: It's a bit cumbersome, I know, but it should do the job :wink:

**Pull request merged**